bokehjs/src/coffee/models/ranges/data_range1d.coffee

```diff
@@ -125,9 +136,17 @@ export class DataRange1d extends DataRange
     [start, end] = @_compute_range(min, max)
 
     if @_initial_start?
-      start = @_initial_start
+      if @_mapper_type == "log"
+        if @_initial_start > 0
+          start = @_initial_start
+      else
+        start = @_initial_start
     if @_initial_end?
-      end = @_initial_end
+      if @_mapper_type == "log"
+        if @_initial_end > 0
+          end = @_initial_end
+      else
+        end = @_initial_end
```

Comment:
In my experience using matplotlib and matlab, plotting data on a log scale does silently ignore values less than zero.  This should be understood as part of the logarithm function.